Remove some unnecessary <iostream> includes from headers

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #106915
* __->__ #106914

In almost all cases this is only included for writing the output formatter, which
only uses `std::ostream` so including `<ostream>` is sufficient.

The istream header is ~1000 lines so the difference is non-trivial.

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @mcarilli @ptrblck @leslie-fang-intel @EikanWang @voznesenskym @penguinwu @anijain2305 @Guobing-Chen @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @aakhundov

peterbell10 committed:
Remove some unnecessary <iostream> includes from headers

In almost all cases this is only included for writing the output formatter, which
only uses `std::ostream` so including `<ostream>` is sufficient.

The istream header is ~1000 lines so the difference is non-trivial.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove some unnecessary <iostream> includes from headers"


In almost all cases this is only included for writing the output formatter, which
only uses `std::ostream` so including `<ostream>` is sufficient.

The istream header is ~1000 lines so the difference is non-trivial.

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 mcarilli ptrblck leslie-fang-intel EikanWang voznesenskym penguinwu anijain2305 Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy chenyang78 aakhundov

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove some unnecessary <iostream> includes from headers"


In almost all cases this is only included for writing the output formatter, which
only uses `std::ostream` so including `<ostream>` is sufficient.

The istream header is ~1000 lines so the difference is non-trivial.

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 mcarilli ptrblck leslie-fang-intel EikanWang voznesenskym penguinwu anijain2305 Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy chenyang78 aakhundov

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove some unnecessary <iostream> includes from headers"


In almost all cases this is only included for writing the output formatter, which
only uses `std::ostream` so including `<ostream>` is sufficient.

The istream header is ~1000 lines so the difference is non-trivial.

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 mcarilli ptrblck leslie-fang-intel EikanWang voznesenskym penguinwu anijain2305 Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy chenyang78 aakhundov

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove some unnecessary <iostream> includes from headers"


In almost all cases this is only included for writing the output formatter, which
only uses `std::ostream` so including `<ostream>` is sufficient.

The istream header is ~1000 lines so the difference is non-trivial.

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 mcarilli ptrblck leslie-fang-intel EikanWang voznesenskym penguinwu anijain2305 Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy chenyang78 aakhundov

[ghstack-poisoned]

peterbell10 commented on .github/ci_commit_pins/xla.txt:
```diff
@@ -1 +1 @@
-709c75a04d9b35d83a9509e1534a8aa2046b8912
+c3c16ccac41cb2db6ba88fb31342f4af62c7e15a
```
This is so we include pytorch/xla#5434

lezcano reviewed (approved):
Did you apply these by hand, or is it possible to automatise these?

lezcano commented:
Talking from memory here, but I think that include-what-you-use does do the iosfwd checks.

lezcano commented:
It seems that it suggests the right thing... sometimes https://github.com/include-what-you-use/include-what-you-use/issues/683

ZainRizvi commented:
@pytorchbot revert -m "Sorry, but this is breaking internal builds. Seems like a lot of internal code depends on some of the removed imports" -c ghfirst

kit1980 commented:
@peterbell10 I'll help with relanding this.

peterbell10 committed:
Rebase on "Remove some unnecessary <iostream> includes from headers"


In almost all cases this is only included for writing the output formatter, which
only uses `std::ostream` so including `<ostream>` is sufficient.

The istream header is ~1000 lines so the difference is non-trivial.

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 mcarilli ptrblck leslie-fang-intel EikanWang voznesenskym penguinwu anijain2305 Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy chenyang78 aakhundov

[ghstack-poisoned]

kit1980 commented:
@kit1980 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D48670403).

kit1980 commented:
@kit1980 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D48670403).

kit1980 commented:
@kit1980 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D48670403).

kit1980 commented:
@kit1980 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D48670606).

izaitsevfb commented:
@pytorchbot revert -m 'Causing metal breakage internally, see D48709279' -c ghfirst

kit1980 commented:
Looks like I missed some issues when re-landing first time...

peterbell10 commented:
My branches seem to have been deleted, so I've opened a new stack starting at #108149. Also switched the PR order so hopefully the pch update doesn't need to be reverted again.
